flexmeasures/api/v3_0/users.py

```diff
@@ -91,26 +106,69 @@ def index(self, account: Account, include_inactive: bool = False):
         :status 403: INVALID_SENDER
         :status 422: UNPROCESSABLE_ENTITY
         """
-
         if account is not None:
             check_access(account, "read")
             accounts = [account]
         else:
-            accounts = []
-            for account in db.session.scalars(select(Account)).all():
-                try:
-                    check_access(account, "read")
-                    accounts.append(account)
-                except (Forbidden, Unauthorized):
-                    pass
-
-        users = []
-        for account in accounts:
-            users += get_users(
-                account_name=account.name,
-                only_active=not include_inactive,
+            accounts = get_accessible_accounts()
+        filter_statement = UserModel.account_id.in_([a.id for a in accounts])
+        if include_inactive is False:
+            filter_statement = and_(filter_statement, UserModel.active.is_(True))
+
+        if filter:
+            filter_statement = and_(
```

Comment:
Let's use the same approach as with assets and tokenize the search terms.

Please use `SearchFilterField` and make your own query utility which mostly needs this kind of logic:

```
filter_statement = filter_statement & and_(
            *(
                or_(
                    GenericAsset.name.ilike(f"%{term}%"),
                    Account.name.ilike(f"%{term}%"),
                )
                for term in search_terms
            )
        )
```

Reply:
I’ve implemented this based on the asset API as a reference. There might be a few things I’ve overlooked, so feel free to review it and let me know if anything needs adjustment.